packages/ui-components/src/ExpandableComponent.tsx

```diff
@@ -57,112 +57,88 @@ export interface IExpandableComponentProps {
   onMouseDown?: (e: React.MouseEvent<HTMLSpanElement, MouseEvent>) => any;
 }
 
-export interface IExpandableComponentState {
-  expanded: boolean;
-}
-
 /**
  * A React component for expandable containers.
  */
-export class ExpandableComponent extends React.Component<
-  IExpandableComponentProps,
-  IExpandableComponentState
-> {
-  constructor(props: IExpandableComponentProps) {
-    super(props);
-    this.state = { expanded: false };
-  }
+export const ExpandableComponent: React.FC<IExpandableComponentProps> = props => {
+  const [expanded, setExpandedValue] = React.useState(false);
 
-  toggleDetailsDisplay(): void {
+  const toggleDetailsDisplay = (): void => {
     // Switch expanded flag
-    const newExpandFlag = !this.state.expanded;
-    if (this.props.onBeforeExpand) {
-      this.props.onBeforeExpand(newExpandFlag);
+    const newExpandFlag = !expanded;
+    if (props.onBeforeExpand) {
+      props.onBeforeExpand(newExpandFlag);
     }
-    this.setState({ expanded: newExpandFlag });
-  }
+    setExpandedValue(newExpandFlag);
+  };
 
-  componentDidUpdate(): void {
-    if (this.props.onExpand) {
-      this.props.onExpand(this.state.expanded);
-    }
-  }
+  React.useEffect((): void => {
+    props.onExpand && props.onExpand(expanded);
+  });
 
-  render(): React.ReactElement {
-    const buttonClasses = [ELYRA_BUTTON_CLASS, BUTTON_CLASS].join(' ');
-    const actionButtons = this.props.actionButtons || [];
+  const buttonClasses = [ELYRA_BUTTON_CLASS, BUTTON_CLASS].join(' ');
+  const actionButtons = props.actionButtons || [];
```

Comment:
this should be changed to using nullish coalescing:
```
const actionButtons = props.actionButtons ?? [];
```

but in this case if it's destructured as a prop above we can assign a default like so:
```ts
const Component: React.FC<Props> = ({ actionButtons = [] }) => {
}
```